src/function/gds/gds_frontier.cpp

```diff
@@ -41,6 +41,48 @@ bool FrontierMorselDispatcher::getNextRangeMorsel(FrontierMorsel& frontierMorsel
     return true;
 }
 
+/**
+ * Constructor of ComponentIDs. For Weakly Connected Components:
+* The frontier will always be the entire set of nodes
+* Each node will have its own unique beginning ID
+*Each node starts in its own component group. As the algortihm runs, adjacent nodes will both take smaller the componentID,
+* Which puts them into the same component group.
+ */
+ComponentIDs::ComponentIDs(const common::table_id_map_t<common::offset_t>& numNodesMap_, storage::MemoryManager* mm) {
+    curIter.store(0);
+    for (const auto& [tableID, numNodes] : numNodesMap_) {
+        numNodesMap[tableID] = numNodes;
+        auto memBuffer = mm->allocateBuffer(false, numNodes * sizeof(std::atomic<uint64_t>));
+        std::atomic<uint16_t>* memBufferPtr = 
+            reinterpret_cast<std::atomic<uint16_t>*>(memBuffer.get()->getData());
+
+        // Cast a unique number to each node
+        for (uint64_t i = 0; i < numNodes; ++i) {
```

Comment:
I think that with WCC, this should become uint64_t, since each node will start off with an unique ComponentID